File: algorithms/0461.hamming_distance/hamming_distance.cpp
class Solution {
public:
    int hammingDistance(int x, int y) {
        int distance = 0;
        while(x||y){
            bool result1 = x & 0x01;
            bool result2 = y & 0x01;
            if(result1 != result2){
                ++distance;
            }
            x >>= 1;
            y >>= 1;
        }
        return distance;        
    }
};

File: algorithms/0461.hamming_distance/hamming_distance.rs
impl Solution {
    pub fn hamming_distance(x: i32, y: i32) -> i32 {
        let mut number = x ^ y;
        let mut distance = 0;
        while number > 0 {
            number = number & (number - 1);
            distance += 1;
        }
        distance
    }
}